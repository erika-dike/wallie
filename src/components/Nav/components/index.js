export { default as LinkWithNavItem } from './LinkWithNavItem/LinkWithNavItem';
export { default as LoginNavItem } from './LoginNavItem/LoginNavItem';
