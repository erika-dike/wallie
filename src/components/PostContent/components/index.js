export {
  default as PostMediaContainer,
} from './PostMediaContainer/PostMediaContainer';
export {
  default as PostTextContainer,
} from './PostTextContainer/PostTextContainer';
