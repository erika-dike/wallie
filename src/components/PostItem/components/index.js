export { default as DeletePostModal } from './DeletePostModal/DeletePostModal';
export { default as PostItemMenu } from './PostItemMenu/PostItemMenu';
export { default as StreamItemHeaderTitle } from './StreamItemHeaderTitle/StreamItemHeaderTitle';
export { default as StreamItemHeaderTimeDisplay } from './StreamItemHeaderTimeDisplay/StreamItemHeaderTimeDisplay';
