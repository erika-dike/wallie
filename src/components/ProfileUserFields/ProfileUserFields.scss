.Profile-user-fields {
    .u-text-truncate {
        overflow: visible !important;
        white-space: normal !important;
    }
}