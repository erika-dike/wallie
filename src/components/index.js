export { default as Nav } from './Nav/Nav';
export { default as FieldGroup } from './FieldGroup/FieldGroup';
