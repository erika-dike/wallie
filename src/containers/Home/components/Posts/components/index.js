export { default as PostsCreate } from './PostsCreate/PostsCreate';
export { default as PostsList } from './PostsList/PostsList';
