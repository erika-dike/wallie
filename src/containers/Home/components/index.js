export { default as ProfileCard } from './ProfileCard/ProfileCard';
export { default as TopPosts } from './TopPosts/TopPosts';
export { default as AboutUs } from './AboutUs/AboutUs';
