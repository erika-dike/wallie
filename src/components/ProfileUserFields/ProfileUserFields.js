import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import { UserFields } from '../../components';

import './ProfileUserFields.css';


const ProfileUserFields = ({ profile }) =>
  <div className="Profile-user-fields">
    <UserFields user={profile.user} />
    <div className="Profile-join-date">
      <span className="Icon">
        <i className="fa fa-calendar" aria-hidden="true" />
      </span>
      <span className="Profile-join-date-text u-dir">
        Joined {moment(profile.user.date_created).format('Do MMMM YYYY')}
      </span>
    </div>
  </div>;

ProfileUserFields.propTypes = {
  profile: PropTypes.shape({
    user: PropTypes.shape({
      username: PropTypes.string,
      first_name: PropTypes.string,
      last_name: PropTypes.string,
      email: PropTypes.string,
      num_posts: PropTypes.number,
    }),
    about: PropTypes.string,
    profile_pic: PropTypes.string,
  }).isRequired,
};

export default ProfileUserFields;
