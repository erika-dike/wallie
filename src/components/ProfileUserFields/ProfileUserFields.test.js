import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { UserFields } from '../../components';

import ProfileUserFields from './ProfileUserFields';


describe('ProfileUserFields Component Test Suite', () => {
  let props;

  beforeEach(() => {
    props = {
      profile: {
        user: {
          username: 'john_doe',
          first_name: 'john',
          last_name: 'doe',
          email: 'john_doe@wallie.com',
          num_posts: 12,
          date_created: '2017-05-03T11:12:12.348206Z',
        },
        about: 'engineer @ wallie',
        profile_pic: 'http://fake-pic.jpg',
      },
    };
  });

  describe('base test', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<ProfileUserFields {...props} />);
    });

    it('renders without crashing', () => {
      expect(wrapper).toBeDefined();
    });

    test('props passed in is equal to props possessed', () => {
      expect(Object.keys(wrapper.instance().props)).toEqual(Object.keys(props));
    });

    it('passes the right props to UserFields', () => {
      const userFields = wrapper.find(UserFields);
      expect(userFields).toHaveLength(1);
      expect(userFields.prop('user')).toEqual(wrapper.instance().props.profile.user);
    });
  });

  describe('Snapshot testing', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<ProfileUserFields {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
