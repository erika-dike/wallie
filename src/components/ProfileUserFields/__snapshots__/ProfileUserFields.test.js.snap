exports[`ProfileUserFields Component Test Suite Snapshot testing renders correctly 1`] = `
<div
  className="Profile-user-fields">
  <div
    className="user-fields account-group">
    <div
      className="name u-text-truncate text-capitalize">
      <a
        className="u-text-inherit-color"
        href={null}
        onClick={null}
        rel="noopener">
        John Doe
      </a>
      <span
        className="user-badges" />
    </div>
    <span
      className="screenname u-inline-block u-dir"
      dir="ltr">
      <a
        className="screename-link u-link-complex u-link-clean"
        href={null}
        onClick={null}
        rel="noopener">
        <span
          className="username u-dir"
          dir="ltr">
          @
          <b
            className="u-link-complex-target">
            john_doe
          </b>
        </span>
      </a>
    </span>
  </div>
  <div
    className="Profile-join-date">
    <span
      className="Icon">
      <i
        aria-hidden="true"
        className="fa fa-calendar" />
    </span>
    <span
      className="Profile-join-date-text u-dir">
      Joined 
      10th May 2017
    </span>
  </div>
</div>
`;
